Tighten the ROM bad call results

fdupress opened the pull request:
This strengthens the generic up to bad results on random oracles.

Any proof that was using the looser bound can be recovered easily with `smt().`

bgregoir reviewed (commented):
Sorry, I did not see this one before. 
What is the interest of breaking proofs ? 
I think the good way to go, is to add new lemmas using new names and to 
add to prove the old lemma with the introduced one.

fdupress commented:
@bgregoir As mentioned on Friday, I just saw that you'd commented.
I've rebased the commit and addressed your comment.

If you're happy, we should merge this even if we plan on later cleaning up the PROM + ROM mess.
